src/mongocrypt-kek.c

```diff
@@ -225,6 +263,10 @@ _mongocrypt_kek_copy_to (const _mongocrypt_kek_t *src, _mongocrypt_kek_t *dst)
          bson_strdup (src->provider.gcp.key_version);
       dst->provider.gcp.endpoint =
          _mongocrypt_endpoint_copy (src->provider.gcp.endpoint);
+   } else if (src->kms_provider == MONGOCRYPT_KMS_PROVIDER_KMIP) {
+      dst->provider.kmip.endpoint =
+         _mongocrypt_endpoint_copy (src->provider.kmip.endpoint);
+      dst->provider.kmip.key_id = bson_strdup (src->provider.kmip.key_id);
```

Comment:
We don't have to check for the presence of key_id here?

Reply:
No. `bson_strdup` returns `NULL` if the argument is `NULL`: http://mongoc.org/libbson/current/bson_strdup.html#description